### PR TITLE
[WIP] Adding BoostedRDNRegressor for regression task

### PR DESCRIPTION
**Pull request by harshakokel**
Adding support for the [BoostSRL regression task](https://starling.utdallas.edu/software/boostsrl/wiki/regression/).

I have added the BoostedRDNRegressor class in the `srlearn.rdn`. This was done to keep the coding style consistent with `sklearn.ensemble.gradient_boosting` which has `sklearn.ensemble.GradientBoostingRegressor` and `sklearn.ensemble.GradientBoostingClassifier` in same file.  I have also added an example of housing price prediction with [Boston housing dataset](https://starling.utdallas.edu/datasets/boston-housing/) in the comments. 

I think the current merge is functionally enough, but below is a list of tasks that remain pending. @hayesall, please let me know if there are more things to be done before the merge. Also, let me know if the class name or location of the class needs to change. 

Pending tasks
[ ] Add test cases for Regression
[ ] Test if comment shows up in the document properly
[ ] Remove redundant negative example file
[ ] Can also remove redundant predicate `regressionExample` and come up with a sklearn way of providing true values.

### CODE REVIEW

**Review by hayesall — APPROVED** [review]
Looks good! See the review comments, and I'll merge this.

Your Pending Tasks bring up some good points.

I don't think all of them need to be done right now, but some feedback is below.

---

> Test if comment shows up in the document properly

Add `rdn.BoostedRDNRegressor` to the [`docs/source/api.rst`](https://github.com/hayesall/srlearn/blob/master/docs/source/api.rst)

> Add test cases for Regression

I ran this a few times, and it looks like Regression is still difficult to reproduce exactly. A few thoughts:

1. Maybe [`assert_array_almost_equal`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_almost_equal.html#numpy-testing-assert-array-almost-equal) within 5-10?
2. There's an open problem in SRLBoost for setting random seeds to reproduce results exactly ([see comment here](https://github.com/hayesall/SRLBoost/blob/97409a1201e007a44292a9854f1a4a46607866be/src/main/java/edu/wisc/cs/will/Boosting/RDN/WILLSetup.java#L68-L77))

> Remove redundant negative example file

Good point. This is more a weakness of the `Database.from_files` method. Initializing a `Database` object directly seems cleaner, e.g.:

```python
db = Database()
db.pos = [ ... ]
db.facts = [ ... ]
```

> Can also remove redundant predicate `regressionExample` and come up with a `sklearn` way of providing true values.

Also agree. I'd have to think through "*how I want to use this*" first. If you're implementing something and think of a better way to express this let me know!

File: srlearn/rdn.py [hunk 1]
```diff
@@ -279,3 +279,266 @@ def predict_proba(self, database):
...
+
+        # Collect the classifications.
+        _results_db = self.file_system.files.TEST_DIR.value.joinpath(
+            "results_" + self.target + ".db"
+        )
+        _pred,  _true = np.loadtxt(
+            _results_db,
+            delimiter="\t",
+            usecols=(1,2),
+            converters={0: lambda s: 0 if s[0] == 33 else 1},
```
[the remark this answers is not in the document]
Comment (harshakokel) [reply srlearn/rdn.py]: resolved.

File: srlearn/rdn.py [hunk 2]
```diff
@@ -279,3 +279,266 @@ def predict_proba(self, database):
...
+
+    >>> from srlearn.rdn import BoostedRDNRegressor
+    >>> from srlearn import Background
+    >>> from srlearn import Database
+    >>> train = Database.from_files(
+         pos="./datasets/Boston/train/pos.pl",
+            neg="./datasets/Boston/train/neg.pl",
+            facts="./datasets/Boston/train/facts.pl",
+            lazy_load=False
+        )
```
[the remark this answers is not in the document]
Comment (harshakokel) [reply srlearn/rdn.py]: resolved

**Comment by harshakokel** [comment]
Added Boston dataset and resolved the review comments comment

**Pull request merged**